pkg/apis/networking/validation/validation.go

```diff
@@ -312,8 +315,9 @@ func ValidateIngressCreate(ingress *networking.Ingress) field.ErrorList {
 func ValidateIngressUpdate(ingress, oldIngress *networking.Ingress) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMetaUpdate(&ingress.ObjectMeta, &oldIngress.ObjectMeta, field.NewPath("metadata"))
 	opts := IngressValidationOptions{
-		AllowInvalidSecretName:       allowInvalidSecretName(oldIngress),
-		AllowInvalidWildcardHostRule: allowInvalidWildcardHostRule(oldIngress),
+		AllowInvalidSecretName:              allowInvalidSecretName(oldIngress),
+		AllowInvalidWildcardHostRule:        allowInvalidWildcardHostRule(oldIngress),
+		AllowedRelaxedServiceNameValidation: allowRelaxedServiceNameValidation(oldIngress),
```

Comment:
I would just set this always `true` for Updates and `false` for Creates.

Also, the name is misleading because you obey the feature gate too... I'd make it something like `SkipServiceNameValidation` (and then just skip validation)

Reply:
> I would just set this always true for Updates and false for Creates.

Could you explain the logic behind this? Not sure I understand

Reply:
Given [this discussion](https://github.com/kubernetes/enhancements/pull/5315#discussion_r2133111930), I had assumed that what I have implemented in this PR was enough.

I'm happy to change it to be what you've suggested here, that doesn't seem too difficult.